Marlin/src/module/stepper_indirection.cpp

```diff
@@ -139,7 +139,27 @@
   }
 #endif // TMC26X
 
-#if HAS_TRINAMIC
+#if STEALTHCHOP_ENABLED
+  static constexpr bool stealthChop_by_axis[] = {
+    #if ENABLED(STEALTHCHOP_XY)
+      true
+    #else
+      false
+    #endif
+    ,
+    #if ENABLED(STEALTHCHOP_Z)
+      true
+    #else
+      false
+    #endif
+    ,
+    #if ENABLED(STEALTHCHOP_E)
+      true
+    #else
+      false
+    #endif
+  };
+  enum StealthIndex : uint8_t { STEALTH_AXIS_XY, STEALTH_AXIS_Z, STEALTH_AXIS_E };
   #define _TMC_INIT(ST, SPMM_INDEX, STEALTH_INDEX) tmc_init(stepper##ST, ST##_CURRENT, ST##_MICROSTEPS, ST##_HYBRID_THRESHOLD, planner.settings.axis_steps_per_mm[SPMM_INDEX], stealthChop_by_axis[STEALTH_INDEX])
```

Comment:
Since this requires the `stealthChop_by_axis` array, which only exists if `STEALTHCHOP_ENABLED` we can assume that `_TMC_INIT` also requires `STEALTHCHOP_ENABLED`. This implies that `AXIS_IS_TMC(...)` for any axis also necessitates and assumes `STEALTHCHOP_ENABLED` to be enabled.

Reply:
`stealthChop_by_axis` should be defined whenever there's a TMC driver (`HAS_TRINAMIC`).